### PR TITLE
SelfSubjectRulesReview test coverage

### PR DESCRIPTION
**Pull request by Skyterix1991**

### CODE REVIEW

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java [hunk 1]
```diff
@@ -44,19 +48,100 @@
...
+    createSubjectAccessReviewTest();
+
+    SelfSubjectRulesReview selfSubjectRulesReview = new SelfSubjectRulesReviewBuilder()
+            .withNewSpec()
+            .withNamespace("test")
+            .endSpec()
+            .withNewStatus()
+            .withResourceRules(new ResourceRuleBuilder()
+                    .withResourceNames("pod")
+                    .withVerbs("get").build()).endStatus().build();
```
Comment (kunal-kushwaha) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java]: Remove duplicate code, you can write these tests in the same function.
Maybe something like createSelfSubjectRulesReviewTest()
Don't delete the existing test createSubjectAccessReviewTest()

**Comment by kunal-kushwaha** [comment]
LGTM

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright (C) 2015 Red Hat, Inc.
- *
+ * <p>
```
Comment (rohanKanojia) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java]: Could you please fix license headers by running: `mvn -N license:format`

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java [hunk 3]
```diff
@@ -38,83 +42,112 @@
...
+
+        SelfSubjectRulesReview rulesReviewResponse = client.subjectAccessReviewAuth().list().create(selfSubjectRulesReview);
+
+        assertNotNull(rulesReviewResponse);
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        SelfSubjectRulesReview reviewFromFile = null;
+
+        try {
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java]: This is a unit test, you shouldn't be catching exceptions (add throws Exception to method signature)

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java [hunk 4]
```diff
@@ -38,83 +42,112 @@
...
-    assertEquals(true, reviewResponse.getStatus().getAllowed());
-
-  }
-}
+    private static final Logger logger = LoggerFactory.getLogger(ClusterRoleBindingCrudTest.class);
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    public void createSubjectRulesReviewTest() {
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java]: This test is testing 2 different scenarios, creating a self subject rule review programmatically (using DSL), and via deserialization. You should separate the test into 2 different tests (methods), one for each scenario.

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java [hunk 5]
```diff
@@ -38,83 +42,112 @@
...
+                .withNewStatus()
+                .withResourceRules(new ResourceRuleBuilder()
+                        .withResourceNames("pod")
+                        .withVerbs("create", "list", "get", "delete").build()).endStatus().build();
+
+        SelfSubjectRulesReview rulesReviewResponse = client.subjectAccessReviewAuth().list().create(selfSubjectRulesReview);
+
+        assertNotNull(rulesReviewResponse);
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java]: Kubernetes-client provides different means for Deserialization, please either use Serialization.unmarshal or KubernetesClient#load to deserialize the class so the test actually tests something from our codebase.

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**